scripts/IncrementMinorVersion.php

```diff
@@ -0,0 +1,99 @@
+<?php
+/**
+ * Copyright (c) Microsoft Corporation.  All Rights Reserved.
+ * Licensed under the MIT License.  See License in the project root
+ * for license information.
+ * 
+ * Bumps up the minor version in src/Core/GraphConstants.php & README based on the latest published package version on Packagist
+ * 
+ * Assumptions:
+ *  - Script is run from the repo root
+ *  - Script is run on a Unix environment (affects file path separator to files)
+ *  - Packagist returns tagged versions in descending order (latest release first)
+*/
+
+const CONSTANTS_FILEPATH = "./src/Core/GraphConstants.php";
+const SDK_VERSION_VAR_NAME = "SDK_VERSION"; # Name of version variable in GraphConstants.php
+const PACKAGIST_ENDPOINT = "https://packagist.org/packages/microsoft/microsoft-graph.json";
+
+function getLatestPackagistVersion(): string 
+{
+    $handle = curl_init();
+    curl_setopt($handle, CURLOPT_URL, PACKAGIST_ENDPOINT);
+    curl_setopt($handle, CURLOPT_RETURNTRANSFER, true);
+    curl_setopt($handle, CURLOPT_TIMEOUT, 100);
+    curl_setopt($handle, CURLOPT_FAILONERROR, true);
+    
+    echo "Fetching latest SDK version from " . PACKAGIST_ENDPOINT . "\n";
+    $response = curl_exec($handle);
+
+    if (curl_error($handle)) {
+        throw new Exception("Failed to get latest packagist version: ". curl_error($handle));
+    }
+
+    curl_close($handle);
+
+    $responseJson = json_decode($response, true);
+    if (!array_key_exists("package", $responseJson) 
+        || !array_key_exists("versions", $responseJson["package"]) 
+        || empty($responseJson["package"]["versions"])) {
+
+        throw new Exception("Unable to find versions in the packagist response JSON: ". $responseJson);
+    }
+
+    $versions = $responseJson["package"]["versions"];
+    foreach ($versions as $version => $versionMetadata) {
+        # Ignore branch versions
+        if (!preg_match('/^dev-.*/', $version)) {
```

Comment:
The example branch names provided would be listed as `dev-a**`, `dev-b**`, `dev-c**` on Packagist & the current check would handle this scenario.

By default branches are exposed on Packagist as versions by prefixing the branch name with `dev-` or if the branch name matches a version e.g. v1, v2 etc, you can access it by appending `dev`. [(more here)](https://getcomposer.org/doc/articles/versions.md#branches)

I have noted that I'm not covering the second branch naming case where `dev` may be the suffix. I'm considering adding an OR check to handle it.

```php
if (!preg_match('/^dev-.*/', $version) && !preg_match('/[0-9]+\.[0-9]+\.[0-9]+-dev/', $version))
```
I'm opting for this as an easier option than matching the semver form in a way that allows processing of e.g. `1.30.1-preview`, `1.30.1-RC1` but skips `2.0.0-dev`. The regex seems tricky to me since there's no easy way to do a "not dev prefix" [regex...]-!dev kinda thing. 

Your thoughts?

Reply:
A better regex for the `-dev` suffix would be:
```php
if (!preg_match('/^dev-.*/', $version) && !preg_match('/.+-dev/', $version))
```

Reply:
You are assuming a single branch from dev, not a branch of a branch from dev.

If I branch `aaa` from `dev`, we get `dev-aaa` package. If I branch `bbb` from `aaa`, I believe we will see an `aaa-bbb` package. I think we can verify when you tag the next generated release as that is when the branch-based releases get picked up. I created dev > aaa > bbb branches. Can we get that generated PR pulled in so we can test this?

Reply:
Sure sounds good

Reply:
Released the generation changes. The two branches show as `dev-aaa` and `dev-bbb`. Seems like `dev` is just a prefix to denote any branch version